src/studio-home/hooks.jsx

```diff
@@ -29,10 +31,19 @@ const useStudioHome = () => {
   const isFailedLoadingPage = studioHomeLoadingStatus === RequestStatus.FAILED;
 
   useEffect(() => {
-    dispatch(fetchStudioHomeData(location.search ?? ''));
-    setShowNewCourseContainer(false);
+    if (!isPaginated) {
+      dispatch(fetchStudioHomeData(location.search ?? ''));
+      setShowNewCourseContainer(false);
+    }
   }, [location.search]);
 
+  useEffect(() => {
+    if (isPaginated) {
+      const { currentPage } = studioHomeCoursesParams;
+      dispatch(fetchStudioHomeData(location.search ?? '', false, { page: currentPage }, true));
+    }
+  }, [studioHomeCoursesParams.currentPage]);
```

Comment:
In terms of keeping complexity of new code as low as possible, is this effect really needed? If you look at the react documentation https://react.dev/learn/you-might-not-need-an-effect, a good case for not using an effect is to handle user events. The recommendation is if you want to trigger a fetch based on a user event - in this case, changing the page - to just call the fetch directly from the event handler, not using an effect based on a state update. This makes the code simpler and avoids bugs due to complexity. What do you think?

Reply:
You're right we should reduce the complexity and make this dispatch inside of the on `handlePageSelected` but for instance we will need other deferent params to `currentPage` so we handle them in one place which it would be here like in this other related [PR](https://github.com/openedx/frontend-app-course-authoring/pull/846) and we would need also a useEffect without an array dependency to call the first page. what do you think?

Reply:
Sure, you'd need to use the effect with an empty array for the first page. That's very common. Otherwise, in terms of handling everything in one place, I'd usually use a function that contains all logic that's in the useEffect now. This function can be called from this useEffect, and also from any place that handles user events and wants the same logic. That way you have everything in one place. The difference is that in the first case, you were doing a flow of A) handling a fired user event -> B) setting state -> C) triggering an effect in another component based on a state change -> D) triggering data fetching. This can be shortened to A -> D, incidentally also setting state if needed. The reason that I see this as preferable is that useEffects introduce a lot of complexity and circumvent react state in a way, and it's very hard to know when what state changes. So someone reading the code may not have any means to find this ABCD chain because there's no direct function call from A that triggers D.
Now this is no big deal if this happens once or twice in a codebase but if you follow this pattern often, it can make the logic extremely hard to follow. So if there's no dire need for this solution here, I'd think it would be cleaner to just extract this into a function and call it.

Reply:
Thanks for the suggestion I really appreciate it. Now I'm changing the page inside `handlePageSelected` when a page is selected and not inside the useEffect that I had, the useEffect is now with an empty array at the beginning for the first time and then each page call the this dispatch to bring the page, please let me know if that is okay for you. Thanks a lot